columnflow/tasks/framework/plotting.py

```diff
@@ -0,0 +1,166 @@
+# coding: utf-8
+
+"""
+Base tasks for different types of plotting tasks
+"""
+
+import law
+import luigi
+
+from columnflow.tasks.framework.mixins import PlotMixin, CategoriesMixin, DatasetsProcessesMixin
+from columnflow.util import test_float
+
+
+class PlotBase(PlotMixin):
+    """
+    Base class for all plotting tasks.
+    """
+    # TODO implement in task
+    plot_suffix = luigi.Parameter(
+        default="",
+        significant=True,
+        description="adds a suffix to the output file name of a plot; empty default",
+    )
+    skip_legend = luigi.BoolParameter(
+        default=False,
+        significant=False,
+        description="when True, no legend is drawn; default: False",
+    )
+    skip_cms = luigi.BoolParameter(
+        default=False,
+        significant=False,
+        description="when True, no CMS logo is drawn; default: False",
+    )
+
+    def get_plot_parameters(self):
+        # convert parameters to usable values during plotting
+        return {
+            "skip_legend": self.skip_legend,
+            "skip_cms": self.skip_cms,
+        }
+
+
+class PlotBase1D(PlotBase):
+    """
+    Base class for plotting tasks creating 1-dimensional plots.
+    """
+
+    skip_ratio = luigi.BoolParameter(
+        default=False,
+        significant=False,
+        description="when True, no ratio (usually Data/Bkg ratio) is drawn in the lower panel; "
+        "default: False",
+    )
+    yscale = luigi.ChoiceParameter(
```

Comment:
For consistency, I would still vote for `y_scale`. Below you used `z_scale` already :)

Reply:
We could use y_scale as parameter for the command line and directly switch to `yscale` here:
https://github.com/uhh-cms/columnflow/blob/feature/PlotBase/columnflow/tasks/framework/plotting.py#L73
But I feel like this is much more confusing than just staying with yscale consistently.

Reply:
I think it should be either `yscale` and `zscale`, or `y_scale` and `z_scale`, and I'd have a slight tendency towards the latter with other parameters in mind that might be added in the future, such as "x_range", "x_min", "x_max", ....